libs/sdk-core/src/swap_in/swap.rs

```diff
@@ -61,19 +62,25 @@ impl SwapperAPI for BreezServer {
         let req = GetSwapPaymentRequest {
             payment_request: bolt11,
         };
-        self.get_fund_manager_client()
+        let resp = self
+            .get_fund_manager_client()
             .await?
             .get_swap_payment(req)
             .await?
             .into_inner();
-        Ok(())
+
+        match resp.swap_error() {
+            crate::grpc::get_swap_payment_reply::SwapError::NoError => Ok(()),
+            err => Err(anyhow!("Failed to complete swap: {}", err.as_str_name())),
```

Comment:
Good catch! Probably that means that we didn't catch these errors before your fix.

Reply:
Thanks. Yes, i's now propagating the swapper error, if any, all the way to `last_redeem_error`. Before they were silent errors.